expression/builtin_encryption.go

```diff
@@ -112,9 +130,42 @@ func (b *builtinAesDecryptSig) evalString(row chunk.Row) (string, bool, error) {
 		return "", true, errors.Trace(err)
 	}
 
-	// TODO: Support other modes.
-	key := encrypt.DeriveKeyMySQL([]byte(keyStr), aes128ecbBlobkSize)
-	plainText, err := encrypt.AESDecryptWithECB([]byte(cryptStr), key)
+	var iv string
+	modeName, _ := b.ctx.GetSessionVars().GetSystemVar(variable.BlockEncryptionMode)
+	mode, exists := aesModes[strings.ToLower(modeName)]
+	if !exists {
+		return "", true, errors.Errorf("unsupported block encryption mode - %v", modeName)
+	}
+	if mode.ivRequired {
+		if len(b.args) != 3 {
+			return "", true, ErrIncorrectParameterCount.GenByArgs("aes_decrypt")
+		}
+		iv, isNull, err = b.args[2].EvalString(b.ctx, row)
+		if isNull || err != nil {
+			return "", true, errors.Trace(err)
+		}
+		if len(iv) < aes.BlockSize {
```

Comment:
what about give an `IVSize` to help understand~? although our `IVSize = BlockSize = 16`

Reply:
OK. add const IVSize = aes.BlockSize.